### ISSUE TITLE
Crash when query from ContentProvider across process

### ISSUE THREAD

**Issue by tankery** [agrosner/DBFlow#1514]
DBFlow Version:
4.2.4

Bug or Feature Request:
Bug

Description:
When use Generated ContentProvider to query across process with multiple-modules feature enabled, there is a chance to crash the requester App .

Step to reproduce:
1. Generate ContentProvider for a database in App A with `@ContentProvider` annotation.
2. Generate database using `targetModuleName` for App A.
3. Init `FlowManager` in `Application.onCreate` of App A, with right "ModuleName" specified when config the DatabaseHolder.
4. Make sure App A is not started before queried by ContentProvider.
5. Query model from App B. This will launch the App A to handle the query.
6. App B crashed.

Crash stack of App B:
```
AndroidRuntime: Caused by: java.lang.IllegalStateException: The global database holder is not initialized. Ensure you call FlowManager.init() before accessing the database.
                    at android.os.Parcel.readException(Parcel.java:1950)
                    at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
                    at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
                    at android.content.ContentProviderProxy.query(ContentProviderNative.java:418)
                    at android.content.ContentResolver.query(ContentResolver.java:754)
                    at android.content.ContentResolver.query(ContentResolver.java:704)
                    at android.content.ContentResolver.query(ContentResolver.java:662)
                    at com.raizlabs.android.dbflow.structure.provider.ContentUtils.query(ContentUtils.java:214)
                    at app.b.data.db.DbAccessorImpl.getAppAData(DbAccessorImpl.java:64)
```

Error in App A:
```
DatabaseUtils: Writing exception to parcel
               java.lang.IllegalStateException: The global database holder is not initialized. Ensure you call FlowManager.init() before accessing the database.
                   at com.raizlabs.android.dbflow.config.FlowManager.checkDatabaseHolder(FlowManager.java:507)
                   at com.raizlabs.android.dbflow.config.FlowManager.getDatabase(FlowManager.java:149)
                   at app.a.data.db.AppA_Provider.query(AppA_Provider.java:58)
                   at android.content.ContentProvider.query(ContentProvider.java:1057)
                   at android.content.ContentProvider.query(ContentProvider.java:1149)
                   at android.content.ContentProvider$Transport.query(ContentProvider.java:242)
                   at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:102)
                   at android.os.Binder.execTransact(Binder.java:674)
```

My speculate:
1. Android will do query before `Application.onCreate`, we need to init `FlowManager` by `ContentProvider` itself.
2. `BaseContentProvider` do have an init for `FlowManager` in `onCreate`, but is for general database. If we want to init module database, we should construct BaseContentProvider with `BaseContentProvider(@NonNull Class<? extends DatabaseHolder> databaseHolderClass)`
3. `@ContentProvider` annotation haven't provider a way to specific the module name, and generated `<database>_Provider` is final, we can't extends it to provider `DatabaseHolder` class.

So how can we do to handle this case? Thanks.

**Comment by 10erlosh** [comment]
@tankery did you find a solution for this?

**Comment by tankery** [comment]
@10erlosh There is a workaround fix for this issue, that init `FlowManager` in `Application.attachBaseContext`, This method will called before `ContentProvider` create.

**Comment by agrosner** [comment]
seems like that is the workaround, to call init in `FlowManager.init()` in the `Application.attachBaseContext`. This is fixed in `5.0.0-alpha2`, you can now specify an `initializeHolderClass`, which will pass up the class to the base content provider.
